honeycomb/honeycomb.go

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019, Honeycomb, Hound Technology, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package honeycomb contains a trace exporter for Honeycomb
+package honeycomb
+
+import (
+	"fmt"
+	"github.com/google/uuid"
+	"google.golang.org/grpc/codes"
+	"time"
+
+	libhoney "github.com/honeycombio/libhoney-go"
+	"go.opentelemetry.io/sdk/trace"
+)
+
+// Exporter is an implementation of trace.Exporter that uploads a span to Honeycomb
+type Exporter struct {
+	Builder        *libhoney.Builder
+	SampleFraction float64
+	// Service Name identifies your application. While optional, setting this
+	// field is extremely valuable when you instrument multiple services. If set
+	// it will be added to all events as `service_name`
+	ServiceName string
+}
+
+// Annotation represents an annotation with a value and a timestamp.
+type Annotation struct {
+	Timestamp time.Time `json:"timestamp"`
+	Value     string    `json:"value"`
+}
+
+// Span is the format of trace events that Honeycomb accepts
+type Span struct {
+	TraceID         string       `json:"trace.trace_id"`
+	Name            string       `json:"name"`
+	ID              uint64       `json:"trace.span_id"`
+	ParentID        uint64       `json:"trace.parent_id,omitempty"`
+	DurationMs      float64      `json:"duration_ms"`
```

Comment:
I believe it's idiomatic to use `Milli` c.f. https://golang.org/pkg/time/'s `StampMilli`